pipeline/pipeline.go

```diff
@@ -0,0 +1,88 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pipeline // import "go.opentelemetry.io/collector/pipeline"
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+type ID struct {
+	typeVal component.DataType
+	nameVal string
+}
+
+// Type returns the type of the component.
+func (id ID) Type() component.DataType {
+	return id.typeVal
+}
+
+// Name returns the custom name of the component.
+func (id ID) Name() string {
+	return id.nameVal
+}
+
+// NewID returns a new ID with the given DataType and empty name.
+func NewID(typeVal component.DataType) ID {
+	return ID{typeVal: typeVal}
+}
+
+// NewIDWithName returns a new ID with the given DataType and name.
+func NewIDWithName(typeVal component.DataType, nameVal string) ID {
+	return ID{typeVal: typeVal, nameVal: nameVal}
+}
+
+// MarshalText implements the encoding.TextMarshaler interface.
+// This marshals the type and name as one string in the config.
+func (id ID) MarshalText() (text []byte, err error) {
+	return []byte(id.String()), nil
+}
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface.
+func (id *ID) UnmarshalText(text []byte) error {
```

Comment:
We could have an internal package that has this implementation and allows it to be reused by both `component.ID` and `pipeline.ID`. One way of doing this would be having a function that returns two strings and we can reuse the parsing logic.

Reply:
I don't know if this is actually desirable - the root module (which internal is a part of) already imports component, so this would create a dependency loop.  It shouldn't create an import loop but I'm still not sure if this is desirable.